plugins/parquet/parquetembed.cpp

```diff
@@ -293,6 +321,7 @@ std::shared_ptr<parquet::arrow::RowGroupReader> ParquetReader::queryCurrentTable
         tables += fileTableCounts[i];
         if (currTable < tables)
         {
+            currentTableMetadata = parquetFileReaders[i]->parquet_reader()->metadata()->Subset({static_cast<int>(currTable - offset)});
```

Comment:
Question: Does this assume that a subset of the data is always the leading fields?  I couldn't find the documentation.

Reply:
I'm not sure what you mean by leading fields, but the field information for each subset should be the same. The only thing that could change is the number of rows in the RowGroup.

Reply:
oh ok.  That makes sense.